eng/test-configuration.json

```diff
@@ -0,0 +1,10 @@
+{
+  "version": 1,
+  "defaultOnFailure": "fail",
```

Comment:
What other values are available here?

Reply:
https://github.com/dotnet/arcade/tree/main/src/Microsoft.DotNet.Helix/Sdk#defaultonfailure

`fail` here would retry only if there is a matching rule. My thinking here was that we shouldn't add retries for *any* failure, rather add them specifically for flaky cases. Which are these particular ones right now.